src/main/java/net/jodah/failsafe/RetryPolicy.java

```diff
@@ -227,6 +229,14 @@ public Duration getDelay() {
   }
 
   /**
+   * Returns the function that determines the next delay given
+   * a failed attempt with the given {@link Throwable}.
+   */
+  public CheckedBiFunction<Object, Throwable, Duration> getDelayFunction() {
```

Comment:
Have you considered to give this function its own interface? E.g.

```java
@FunctionalInterface
public interface DelayFunction {
    @Nullable
    Duration calculateDelay(@Nullable Object result, @Nullable Throwable exception);
}
```

Reply:
Is the idea here that `DelayFunction` is more friendly for Java 6/7 users than `CheckedBiFunction`?

Reply:
Yes, that's a good reason, but also: I can't see a compelling need for a delay function to be able to throw a checked exception. I only initially used `CheckedBiFunction` because there was no `BiFunction` in `net.jodah.failsafe.function`. When @whiskeysierra pointed out that rolling a separate `@FunctionalInterface` would give us an unchecked signature, along with the benefits of a more specific method name and a place to put documentation, I embraced that right away.